lang_java/parsing/parser_java.mly

```diff
@@ -826,6 +839,7 @@ statement_expression:
  | class_instance_creation_expression  { $1 }
  /*(* sgrep-ext: to allow '$S;' in sgrep *)*/
  | IDENTIFIER { Flag_parsing.sgrep_guard ((Name (name [Id $1])))  }
+ | typed_metavar { $1 }
```

Comment:
not sure we need this one.

Reply:
We need it so that we can match (String $A)